main/FatesInterfaceMod.F90

```diff
@@ -1944,7 +1973,63 @@ subroutine UpdateFatesRMeansTStep(sites,bc_in)
            call cpatch%tveg24%UpdateRMean(bc_in(s)%t_veg_pa(ifp))
            call cpatch%tveg_lpa%UpdateRMean(bc_in(s)%t_veg_pa(ifp))
            call cpatch%tveg_longterm%UpdateRMean(bc_in(s)%t_veg_pa(ifp))
+           
+           ! Update seedling layer running means of PAR, soil matric potential and 
+           ! moisture deficit days (mdd).
+           n_leaf = maxval(cpatch%ncan(cpatch%ncl_p,:)) ! Calculate the number of leaf layers
```

Comment:
I'm afraid that this will be averaging in zeros, since its using the maximum leaf layer across pfts, some of those PFTs will not have leaves down as far as others.  I think this needs a loop over pft's using the actual %ncan(ncl_p,pft) for each.

Reply:
Thanks for this comment! Just so its clear before I change anything, I would like all seedlings to experience the same light environment regardless of pft. This light environment should basically be the "average light", underneath all pfts, (perhaps similar to how light gets mixed between the canopy layer and the understory layer)? I suppose my current calculation for lai_sun_frac (L1984) sums over the zeros you mentioned and essentially gives the fraction of lai in the sun for the lowest leaf layer of the pft with the most leaf layers in the understory? This number is then used as the weight in the weighted average of direct and diffuse light at the seedling layer (L1994). So what I think you're getting at is that currently what we're calling light in the seedling layer is light at the lowest leaf layer of the pft with the most leaf layers right? If so, that would be the darkest spots of the forest floor when what we really want is to create an average light below all pfts...

So to get what we want, I should actually calculate the pft specific values of new_seedling_layer_par (L1993) and then do another weighted average across pfts where I weight by total pft-level lai in each pft's respective lowest leaf layer? Something like that?